apiary-metastore-events/kafka-metastore-events/kafka-events-common/src/main/java/com/expediagroup/apiary/extensions/events/metastore/common/listener/AbstractMetaStoreEventListener.java

```diff
@@ -0,0 +1,181 @@
+import org.apache.hadoop.hive.metastore.events.AlterPartitionEvent;
+import org.apache.hadoop.hive.metastore.events.AlterTableEvent;
+import org.apache.hadoop.hive.metastore.events.ConfigChangeEvent;
+import org.apache.hadoop.hive.metastore.events.CreateDatabaseEvent;
+import org.apache.hadoop.hive.metastore.events.CreateFunctionEvent;
+import org.apache.hadoop.hive.metastore.events.CreateTableEvent;
+import org.apache.hadoop.hive.metastore.events.DropDatabaseEvent;
+import org.apache.hadoop.hive.metastore.events.DropFunctionEvent;
+import org.apache.hadoop.hive.metastore.events.DropIndexEvent;
+import org.apache.hadoop.hive.metastore.events.DropPartitionEvent;
+import org.apache.hadoop.hive.metastore.events.DropTableEvent;
+import org.apache.hadoop.hive.metastore.events.InsertEvent;
+import org.apache.hadoop.hive.metastore.events.LoadPartitionDoneEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.expediagroup.apiary.extensions.events.metastore.common.event.SerializableListenerEvent;
+import com.expediagroup.apiary.extensions.events.metastore.common.event.SerializableListenerEventFactory;
+import com.expediagroup.apiary.extensions.events.metastore.common.io.MetaStoreEventSerDe;
+import com.expediagroup.apiary.extensions.events.metastore.common.messaging.Message;
+import com.expediagroup.apiary.extensions.events.metastore.common.messaging.MessageTask;
+import com.expediagroup.apiary.extensions.events.metastore.common.messaging.MessageTaskFactory;
+
+public abstract class AbstractMetaStoreEventListener extends MetaStoreEventListener {
+  private static final Logger log = LoggerFactory.getLogger(AbstractMetaStoreEventListener.class);
+
+  private final SerializableListenerEventFactory serializableListenerEventFactory;
+  private final ExecutorService executorService;
+
+  protected AbstractMetaStoreEventListener(
+      Configuration config,
+      SerializableListenerEventFactory serializableListenerEventFactory,
+      ExecutorService executorService) {
+    super(config);
+    log.info("Creating MetaStoreEventListener of class {}", this.getClass().getName());
+    this.serializableListenerEventFactory = serializableListenerEventFactory;
+    this.executorService = executorService;
+  }
+
+  protected abstract MetaStoreEventSerDe getMetaStoreEventSerDe();
+
+  protected abstract MessageTaskFactory getMessageTaskFactory();
+
+  private MessageTask message(Message message) {
+    return new WrappingMessageTask(getMessageTaskFactory().newTask(message));
+  }
+
+  private Message withPayload(SerializableListenerEvent event) throws MetaException {
+    return Message
+        .builder()
+        .database(event.getDatabaseName())
+        .table(event.getTableName())
+        .payload(getMetaStoreEventSerDe().marshal(event))
+        .build();
+  }
+
+  @Override
+  public void onCreateTable(CreateTableEvent tableEvent) throws MetaException {
+    log.info("Create table event received");
```

Comment:
We should either remove all these or drop them to debug otherwise we'll be outputting *loads*.

Reply:
I have personally found these debug statements very useful when you aren't getting any event on the receiver side. Also, these are listener logs, so will be on the Hive Metastore side which perhaps won't be in our control to start in debug mode.

Reply:
Will change these to debug, so that at least we will still see logging in integration tests